api/searchCases/index.ts

```diff
@@ -1,137 +1,171 @@
+                [field]: searchTerm,
+              },
+            }
+          } else {
+            match = {
+              wildcard: {
+                [field]: `*${searchTerm.toLowerCase()}*`,
+              },
+            }
+          }
+        } else {
+          match = {
+            match: {
+              [field]: {
+                operator: 'and',
+                query: searchTerm,
+              },
             },
-        },
-        size,
-        sort,
+          }
+        }
+        matchList.push(match)
+      }
     }
+  }
 
-    return {
-        native_es_query: nativeEsQuery,
-        supplementary_data: ['*'],
-    }
+  nativeEsQuery = {
+    from,
+    query: {
+      bool: {
+        must: matchList,
+      },
+    },
+    size,
+    sort,
+  }
+
+  return {
+    native_es_query: nativeEsQuery,
+    supplementary_data: ['*'],
+  }
 }
 
 function prepareSort(params) {
-    const sortQuery = []
-    if (params.hasOwnProperty('column') && params.hasOwnProperty('order') && params.hasOwnProperty('type')) {
-        let columnName: string
-
-        if (params.column.indexOf('[') === -1 ) {
-            columnName = `data.${params.column}${isKeywordSuffixNeeded(params.column, params.type)}`
-        } else {
-            const mappedName = fieldNameMapper(
-                params.column.replace('[', '').replace(']', '').toLowerCase(),
-                caseMetaDataFiledsMapping
-            )
-            columnName = `${mappedName}${isKeywordSuffixNeeded(mappedName, params.type)}`
-        }
-        const orderDirection = params.order === 0 ? 'ASC' : 'DESC'
-        sortQuery.push(
-            {
-                [columnName]: orderDirection,
-            }
-        )
+  const sortQuery = []
+  if (params.hasOwnProperty('column') && params.hasOwnProperty('order') && params.hasOwnProperty('type')) {
+    let columnName: string
+
+    if (params.column.indexOf('[') === -1) {
+      columnName = `data.${params.column}${isKeywordSuffixNeeded(params.column, params.type)}`
+    } else {
+      const mappedName = fieldNameMapper(
+        params.column.replace('[', '').replace(']', '').toLowerCase(),
+        caseMetaDataFiledsMapping
+      )
+      columnName = `${mappedName}${isKeywordSuffixNeeded(mappedName, params.type)}`
     }
-    return sortQuery
+    const orderDirection = params.order === 0 ? 'ASC' : 'DESC'
+    sortQuery.push(
+      {
+        [columnName]: orderDirection,
+      }
+    )
+  }
+  return sortQuery
 }
 
 function isKeywordSuffixNeeded(columnName, type): string {
-    const types = ['Text', 'TextArea', 'FixedList', 'FixedListEdit', 'MultiSelectList', 'FixedRadioList', 'DynamicList']
-    const specialFields = ['reference', 'jurisdiction', 'state', 'case_type_id']
-    const isText = types.includes(type) || specialFields.includes(columnName)
-    return isText ? '.keyword' : ''
+  const types = ['Text', 'TextArea', 'FixedList', 'FixedListEdit', 'MultiSelectList', 'FixedRadioList', 'DynamicList']
```

Comment:
These should be defined outside of this instance method as they'll be instantiated every time the method is invoked. Not a blocker - and I know you've just changed the indenting - but it's less efficient than it could be.

Reply:
Sure next time I look at it, I will fix it and I didn't modify this code.